packages/hooks/src/useDtraderV2Flag.ts

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from 'react';
+import useIsGrowthbookIsLoaded from './useIsGrowthbookLoaded';
+import { isDTraderV2, routes } from '@deriv/shared';
+import { useDevice } from '@deriv-com/ui';
+import { Analytics } from '@deriv-com/analytics';
+
+const useDtraderV2Flag = () => {
+    const { isGBLoaded: is_growthbook_loaded, isGBAvailable: is_gb_available } = useIsGrowthbookIsLoaded();
+    const load_dtrader_module = is_growthbook_loaded || !is_gb_available;
+
+    const is_dtrader_v2 = isDTraderV2();
+    const { isMobile } = useDevice();
+    const is_feature_flag_active = Boolean(Analytics?.getFeatureValue('dtrader_v2_enabled', false));
+    const is_trade_or_contract_path =
+        location.pathname.startsWith(routes.trade) || location.pathname.startsWith('/contract/');
+    const is_mobile = isMobile;
```

Comment:
Nit in case if you need just to rename it `const { isMobile: s_mobile} = useDevice();`